Tests added in gh-3009 may fail

Presumably it's flaky since the failure wasn't observed in that PR.

https://travis-ci.org/datalad/datalad/jobs/460830538#L2749

```
======================================================================
FAIL: datalad.support.tests.test_annexrepo.test_commit_annex_commit_changed(True,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/tests/utils.py", line 423, in newfunc
    return t(*(arg + (d,)), **kw)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/support/tests/test_annexrepo.py", line 2476, in check_commit_annex_commit_changed
    , untracked=['untracked']
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/tests/utils.py", line 219, in ok_clean_git
    eq_(sorted(index_modified_), sorted(index_modified))
AssertionError: ['tobechanged-annex', 'tobechanged-git', 'willnotgetshort'] != ['tobechanged-annex', 'tobechanged-git']
======================================================================
FAIL: datalad.support.tests.test_annexrepo.test_commit_annex_commit_changed(False,)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/tests/utils.py", line 423, in newfunc
    return t(*(arg + (d,)), **kw)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/support/tests/test_annexrepo.py", line 2476, in check_commit_annex_commit_changed
    , untracked=['untracked']
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/tests/utils.py", line 219, in ok_clean_git
    eq_(sorted(index_modified_), sorted(index_modified))
AssertionError: ['tobechanged-annex', 'tobechanged-git', 'willnotgetshort'] != ['tobechanged-annex', 'tobechanged-git']
```

--- a/datalad/support/annexrepo.py
+++ b/datalad/support/annexrepo.py
@@ -2896,7 +2896,7 @@ class AnnexRepo(GitRepo, RepoInterface):
                 direct_mode = self.is_direct_mode()
                 # we might need to avoid explicit paths
                 files_to_commit = None if direct_mode else files
-                if files and self.config.getint("annex", "version") < 6:
+                if files:
                     # In direct mode, if we commit file(s) they would get
                     # committed directly into git ignoring possibly being
                     # staged by annex.  So, if not all files are committed, and

